### ISSUE TITLE
Unable to add IPv6 to a VPC and subnets in a single operation

### ISSUE THREAD

**Issue by stevenewey** [terraform-providers/terraform-provider-aws#3398]
### Terraform Version
Terraform v0.11.3
+ provider.aws v1.9.0

### Affected Resource(s)
- aws_vpc
- aws_subnet

### Terraform Configuration Files

Before:

```hcl
provider "aws" {
  region = "eu-west-2"
}

resource "aws_vpc" "testvpc" {
  cidr_block = "10.0.0.0/16"
#  assign_generated_ipv6_cidr_block = true
}

resource "aws_subnet" "testsubnet" {
  cidr_block = "${cidrsubnet(aws_vpc.testvpc.cidr_block, 8, 0)}"
#  ipv6_cidr_block = "${cidrsubnet(aws_vpc.testvpc.ipv6_cidr_block, 8, 0)}"
  vpc_id = "${aws_vpc.testvpc.id}"
}
```

After:
```hcl
provider "aws" {
  region = "eu-west-2"
}

resource "aws_vpc" "testvpc" {
  cidr_block = "10.0.0.0/16"
  assign_generated_ipv6_cidr_block = true
}

resource "aws_subnet" "testsubnet" {
  cidr_block = "${cidrsubnet(aws_vpc.testvpc.cidr_block, 8, 0)}"
  ipv6_cidr_block = "${cidrsubnet(aws_vpc.testvpc.ipv6_cidr_block, 8, 0)}"
  vpc_id = "${aws_vpc.testvpc.id}"
}
```

### Debug Output
https://gist.github.com/stevenewey/67041261f92798d27a679dc3d1d35da4

### Expected Behavior
I should be able to enable IPv6 support for a VPC and subnets in a single step.

### Actual Behavior
During `plan`, an error is raised because the the IPv6 block has not been allocated to the VPC yet. However, when applying the second version as new, it can be cleanly applied.

For now, we are working around this by checking for an IPv6 allocation on the VPC before running terraform, and using the AWS CLI tool to add one if one isn't found. At which point, terraform is able to proceed as normal.

### Steps to Reproduce
1. `terraform apply` the first configuration above
2. update to the second configuration with IPv6 enabled
3. `terraform plan`

**Comment by bflad** [comment]
The fix for this to ensure it waits for IPv6 CIDR block association on resource creation has been merged and will release with version 1.43.1 of the AWS provider, likely later today. 👍

**Comment by stevenewey** [comment]
Thanks @bflad!

**Comment by bflad** [comment]
This has been released in [version 1.43.1 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1431-november-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
